plotly/tools.py

```diff
@@ -1443,6 +1449,333 @@ class FigureFactory(object):
     """
 
     @staticmethod
+    def _unlabel_rgb(colors):
+        unlabelled_colors = []
+        for color in colors:
+            str_vals = ''
+            for index in range(len(color)):
+                try:
+                    float(color[index])
+                    str_vals = str_vals + color[index]
+                except ValueError:
+                    if (color[index] == ',') or (color[index] == '.'):
+                        str_vals = str_vals + color[index]
+
+            str_vals = str_vals + ','
+            numbers = []
+            str_num = ''
+            for char in str_vals:
+                if char != ',':
+                    str_num = str_num + char
+                else:
+                    numbers.append(float(str_num))
+                    str_num = ''
+            unlabelled_tuple = (numbers[0], numbers[1], numbers[2])
+            unlabelled_colors.append(unlabelled_tuple)
+
+        return unlabelled_colors
+
+    @staticmethod
+    def _find_intermediate_color(tuple1, tuple2, t):
+        # Given two color tuples (in normalized RGB space)
+        # and a value 0 < t < 1, spit out a color that is
+        # t percent from tuple1 to tuple2.
+        diff_0 = float(tuple2[0] - tuple1[0])
+        diff_1 = float(tuple2[1] - tuple1[1])
+        diff_2 = float(tuple2[2] - tuple1[2])
+
+        new_tuple = (tuple1[0] + t*diff_0,
+                     tuple1[1] + t*diff_1,
+                     tuple1[2] + t*diff_2)
+        return new_tuple
+
+    @staticmethod
+    def _unconvert_from_RGB_255(colors):
```

Comment:
🐄 You might also consider having it handle *one* color iterable at a time, then you'd just use it in a list comprehension:

`blah = [normalize_rgb_color(color) for color in colors]`